scipy/signal/tests/test_signaltools.py

```diff
@@ -919,22 +922,30 @@ def test_mutable_window(self):
         impulse = np.zeros(3)
         window = np.random.RandomState(0).randn(2)
         window_orig = window.copy()
-        signal.resample_poly(impulse, 5, 1, window=window)
-        assert_array_equal(window, window_orig)
+        for padtype in ['constant', 'mean', 'median', 'maximum', 'minimum', 'line']:
+            signal.resample_poly(impulse, 5, 1, window=window, padtype=padtype)
+            assert_array_equal(window, window_orig)
 
     def test_output_float32(self):
         # Test that float32 inputs yield a float32 output
         x = np.arange(10, dtype=np.float32)
         h = np.array([1, 1, 1], dtype=np.float32)
-        y = signal.resample_poly(x, 1, 2, window=h)
-        assert_(y.dtype == np.float32)
-
-    @pytest.mark.parametrize('method, ext', [
-        ('fft', False),
-        ('polyphase', False),
-        ('polyphase', True),
+        for padtype in ['constant', 'mean', 'median', 'maximum', 'minimum', 'line']:
+            y = signal.resample_poly(x, 1, 2, window=h, padtype=padtype)
+            assert_(y.dtype == np.float32)
+
+    @pytest.mark.parametrize('method, ext, padtype', [
+        ('fft', False, None),
+        ('polyphase', False, 'constant'),
+        ('polyphase', True, 'constant'),
+        ('polyphase', False, 'mean'),
+        ('polyphase', True, 'mean'),
+        ('polyphase', False, 'median'),
+        ('polyphase', True, 'median'),
+        ('polyphase', False, 'line'),
+        ('polyphase', True, 'line'),
```

Comment:
Should 'minimum' and 'maximum' also be here?

anticipating more edge modes may be added in the future it could be worth writing this using `product`:
```python
from itertools import product

@pytest.mark.parametrize(
    "method, ext, padtype",
    [("fft", False, None)]
    + list(
        product(
            ["polyphase"],
            [False, True],
            ["constant", ",mean", "median", "minimum", "maximum", "line"],
        )
    ),
)
```

Reply:
This has not been addressed yet